### PR TITLE
fix(embodied): fix function of ordered reset id; make gradient ckpt into optional; override get_benchmark to support multi-task training;

### PR DESCRIPTION
**Pull request by WinstonWmj**
This PR modifies several LIBERO-related components, including:

* **Bug fix** for the ordered reset ID function in LIBERO.
* **Override** of `get_benchmark()` in the LIBERO source to enable **multi-task training on LIBERO-130** within the RLinF framework.
* **Feature addition:** made `module.gradient_checkpointing_enable()` configurable to allow optional activation of gradient checkpointing.
* **Add ci-test for libero130:** add models for testing`libero` simulator, `openvla-oft` model

### Description
fix function of ordered reset id; make gradient ckpt into optional; override get_benchmark to support multi-task training;

### Motivation and Context
* The LIBERO-related updates represent a major algorithmic improvement, making our framework the **first to support simultaneous training across all LIBERO tasks** with strong performance.
* The new `gradient_checkpointing_enable` option adds **flexibility** to the framework, enabling better memory–compute trade-offs during large-scale training.

### How has this been tested?
Verified stable performance and consistent training curves across **single-node**, **dual-node**, and **four-node** setups, with no observed issues.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rlinf/envs/libero/utils.py [hunk 1]
```diff
@@ -447,3 +447,55 @@ def save_rollout_video(
...
+
+        # Monkey-patch get_benchmark to intercept only when name is libero_130
+        def _get_benchmark_overridden(benchmark_name):
+            if str(benchmark_name).lower() == "libero_130":
+                return LIBERO_ALL
+            return _orig_get_benchmark(benchmark_name)
+
+        bm.get_benchmark = _get_benchmark_overridden
+
+    except Exception:
```
Comment (zanghz21) [inline rlinf/envs/libero/utils.py]: We need Error if LIBERO is not installed. So please throw error in the Exception part, or just del the try-except structure.

Comment (WinstonWmj) [reply rlinf/envs/libero/utils.py]: Useful suggestion! I have removed the try-catch structure.

File: rlinf/envs/libero/libero_env.py [hunk 2]
```diff
@@ -20,11 +20,12 @@
 import numpy as np
 import torch
 from libero.libero import get_libero_path
-from libero.libero.benchmark import Benchmark, get_benchmark
+from libero.libero.benchmark import Benchmark
 from libero.libero.envs import OffScreenRenderEnv
 from omegaconf.omegaconf import OmegaConf
 
 from rlinf.envs.libero.utils import (
+    _get_benchmark_overridden,
```
Comment (QuanluZhang) [inline rlinf/envs/libero/libero_env.py]: it is a little strange that you import a private function (headed with `_`)

File: rlinf/envs/libero/utils.py [hunk 3]
```diff
@@ -447,3 +447,54 @@ def save_rollout_video(
...
+
+        # Register for discoverability/help
+        bm.BENCHMARK_MAPPING["libero_130"] = LIBERO_ALL
+
+        _LIBERO_130_CLASS = LIBERO_ALL
+        return _LIBERO_130_CLASS
+
+    except Exception:
+        # Fallback: raise similar to original behavior if import fails
+        raise
```
Comment (QuanluZhang) [inline rlinf/envs/libero/utils.py]: please raise a clear error with error message.

File: rlinf/hybrid_engines/fsdp/fsdp_model_manager.py [hunk 4]
```diff
@@ -78,7 +78,12 @@ def setup_model_and_optimizer(self):
         """Setup model and optimizer."""
         module = self.model_provider_func()
 
-        module.gradient_checkpointing_enable()
+        # Enable gradient checkpointing if configured
+        if self._cfg.model.get("enable_gradient_checkpointing", True):
+            print("[FSDP] Enabling gradient checkpointing")
+            module.gradient_checkpointing_enable()
+        else:
+            print("[FSDP] Gradient checkpointing is disabled")
```
Comment (QuanluZhang) [inline rlinf/hybrid_engines/fsdp/fsdp_model_manager.py]: please use logging instead of `print`

File: rlinf/hybrid_engines/fsdp/fsdp_model_manager.py [hunk 5]
```diff
@@ -78,7 +78,12 @@ def setup_model_and_optimizer(self):
         """Setup model and optimizer."""
         module = self.model_provider_func()
 
-        module.gradient_checkpointing_enable()
+        # Enable gradient checkpointing if configured
+        if self._cfg.model.get("enable_gradient_checkpointing", True):
```
Comment (QuanluZhang) [inline rlinf/hybrid_engines/fsdp/fsdp_model_manager.py]: it is a little strange to make gradient_checkpointing True by default. And, why not using the same configure field name as pytorch, i.e., `gradient_checkpointing_enable`? It is not user friendly to create a new name `enable_gradient_checkpointing`.

File: rlinf/envs/libero/libero_env.py [hunk 6]
```diff
@@ -154,13 +155,13 @@ def get_reset_state_ids_all(self):
         return reset_state_ids
 
     def _get_ordered_reset_state_ids(self, num_reset_states):
+        if self.start_idx + num_reset_states > len(self.reset_state_ids_all[0]):
+            self.reset_state_ids_all = self.get_reset_state_ids_all()
+            self.start_idx = 0
```
Comment (QuanluZhang) [inline rlinf/envs/libero/libero_env.py:160]: I don't understand this part, @zanghz21 please to double check this part.

File: rlinf/envs/libero/utils.py [hunk 7]
```diff
@@ -447,3 +447,54 @@ def save_rollout_video(
...
+
+_LIBERO_130_CLASS = None
+
+
+def _get_benchmark_overridden(benchmark_name):
+    """
+    Return the Benchmark class for a given name.
+
+    - For "libero_130": return a dynamically aggregated class from all suites.
+    - For others: delegate to the original LIBERO get_benchmark.
```
Comment (QuanluZhang) [inline rlinf/envs/libero/utils.py]: please follow the formal docstring format.

File: rlinf/envs/libero/utils.py (hunk 1 above)
Comment (QuanluZhang) [reply rlinf/envs/libero/utils.py]: have you removed?

Comment (WinstonWmj) [reply rlinf/envs/libero/utils.py]: sry, I removed in [commit](5582caf936aa6fd442da34362fb40d83271174f2)

File: rlinf/envs/libero/libero_env.py (hunk 2 above)
Comment (WinstonWmj) [reply rlinf/envs/libero/libero_env.py]: I renamed it without prefix `_` in [commit](5582caf936aa6fd442da34362fb40d83271174f2)

File: rlinf/envs/libero/utils.py (hunk 3 above)
Comment (WinstonWmj) [reply rlinf/envs/libero/utils.py]: I removed this strucutre since `libero` can raise enough info and error.

File: rlinf/hybrid_engines/fsdp/fsdp_model_manager.py (hunk 4 above)
Comment (WinstonWmj) [reply rlinf/hybrid_engines/fsdp/fsdp_model_manager.py]: okay~

File: rlinf/hybrid_engines/fsdp/fsdp_model_manager.py (hunk 5 above)
Comment (WinstonWmj) [reply rlinf/hybrid_engines/fsdp/fsdp_model_manager.py]: I change this parameter's name into `gradient_checkpointing` and make it **False** in the code by default following `TrainingArguments` in Hugging Face Transformers. However, in order to maintain the experimental results under other settings, I set the value of this parameter to **True** in the YAML files.

File: rlinf/envs/libero/libero_env.py (hunk 6 above)
Comment (WinstonWmj) [reply rlinf/envs/libero/libero_env.py:160]: @XuS1994 reported this bug. After I fixed it, I have run multiple rounds of experiments to verify the correctness of this code segment.

File: rlinf/envs/libero/utils.py (hunk 7 above)
Comment (WinstonWmj) [reply rlinf/envs/libero/utils.py]: Okay

File: rlinf/hybrid_engines/fsdp/fsdp_model_manager.py [hunk 8]
```diff
@@ -78,7 +79,12 @@ def setup_model_and_optimizer(self):
         """Setup model and optimizer."""
         module = self.model_provider_func()
 
-        module.gradient_checkpointing_enable()
+        # Enable gradient checkpointing if configured
+        if self._cfg.model.get("gradient_checkpointing", False):
```
Comment (zanghz21) [inline rlinf/hybrid_engines/fsdp/fsdp_model_manager.py:84]: I think it is better to change to ``self._cfg.model.get("gradient_checkpointing", True)`` here

**Review by XuS1994 — APPROVED** [review]
